instance/service.go

```diff
@@ -0,0 +1,76 @@
+package instance
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+
+	sq "github.com/Masterminds/squirrel"
+	"github.com/influxdata/influxdb/v2"
+	"github.com/influxdata/influxdb/v2/kit/platform"
+	ierrors "github.com/influxdata/influxdb/v2/kit/platform/errors"
+	"github.com/influxdata/influxdb/v2/snowflake"
+	"github.com/influxdata/influxdb/v2/sqlite"
+)
+
+var (
+	errInstanceNotFound = &ierrors.Error{
+		Code: ierrors.ENotFound,
+		Msg:  "instance not found",
+	}
+)
+
+func NewService(store *sqlite.SqlStore) *service {
+	return &service{
+		store:       store,
+		idGenerator: snowflake.NewIDGenerator(),
+	}
+}
+
+type service struct {
+	store       *sqlite.SqlStore
+	idGenerator platform.IDGenerator
+}
+
+func (s service) CreateInstance(ctx context.Context) (*influxdb.Instance, error) {
+	s.store.Mu.Lock()
+	defer s.store.Mu.Unlock()
+
+	q := sq.Insert("instance").
+		SetMap(sq.Eq{
+			"id":         s.idGenerator.ID(),
```

Comment:
Why do we need this, rather than just having the instance be a singleton? the id could always be `1` or something.

Reply:
As discussed offline, we're going to revert to the previous behavior for now since its not necessary for this change.